spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java

```diff
@@ -276,8 +289,25 @@ public void testAnnotatedServiceActivator() {
 		assertFalse(TestUtils.getPropertyValue(trigger, "fixedRate", Boolean.class));
 
 		assertTrue(this.annotationTestService.isRunning());
+		Log logger = spy(TestUtils.getPropertyValue(this.serviceActivatorEndpoint, "logger", Log.class));
+		when(logger.isDebugEnabled()).thenReturn(true);
+		final CountDownLatch pollerInterruptedLatch = new CountDownLatch(1);
+		doAnswer(new Answer<Void>() {
+
+			@Override
+			public Void answer(InvocationOnMock invocation) throws Throwable {
+				pollerInterruptedLatch.countDown();
+				invocation.callRealMethod();
+				return null;
+			}
+		}).when(logger).debug("Received no Message during the poll, returning 'false'");
+		new DirectFieldAccessor(this.serviceActivatorEndpoint).setPropertyValue("logger", logger);
+
 		this.serviceActivatorEndpoint.stop();
 		assertFalse(this.annotationTestService.isRunning());
+
+		// wait until the service activator's poller is interrupted.
+		assertTrue(pollerInterruptedLatch.await(10, TimeUnit.SECONDS));
```

Comment:
It's difficult to me to understand why `this.runningTask.cancel(true);` on the previous `stop()` can steal messages after the next `start()`, but that looks that we should move this "await stop on the start" logic to the `AbstractPollingEndpoint`, because it might be an issue for end applications.
Yes, we can consider that on the next version, but I'd like to hear your point on the matter.

Merging anyway with `LogAdjustingTestSupport` removal from here.
Plus commit message polishing according JIRA link.